src/box/memtx_tree.cc

```diff
@@ -1486,8 +1486,11 @@ memtx_tree_func_index_replace(struct index *base, struct tuple *old_tuple,
 		int err = 0;
 		struct tuple *key;
 		struct func_key_undo *undo;
+		struct key_def *key_def = index_def->key_def;
 		while ((err = key_list_iterator_next(&it, &key)) == 0 &&
```

Comment:
Could you please add an assertion that the `key_def` is never multikey for clarity here and below?

Reply:
Done.
<details>
<summary>diff</summary>

```diff
diff --git a/src/box/memtx_tree.cc b/src/box/memtx_tree.cc
index bcb4555a42..20d4069810 100644
--- a/src/box/memtx_tree.cc
+++ b/src/box/memtx_tree.cc
@@ -1469,6 +1469,8 @@ memtx_tree_func_index_replace(struct index *base, struct tuple *old_tuple,
 		(struct memtx_tree_index<true> *)base;
 	struct index_def *index_def = index->base.def;
 	assert(index_def->key_def->for_func_index);
+	/* Make sure that key_def is not multikey - we rely on it below. */
+	assert(!index_def->key_def->is_multikey);
 
 	int rc = -1;
 	struct region *region = &fiber()->gc;
@@ -1778,6 +1780,8 @@ memtx_tree_func_index_build_next(struct index *base, struct tuple *tuple)
 		(struct memtx_tree_index<true> *)base;
 	struct index_def *index_def = index->base.def;
 	assert(index_def->key_def->for_func_index);
+	/* Make sure that key_def is not multikey - we rely on it below. */
+	assert(!index_def->key_def->is_multikey);
 
 	struct region *region = &fiber()->gc;
 	size_t region_svp = region_used(region);
```
</details>